HACKING.md

```diff
@@ -0,0 +1,66 @@
+# Hacking on Origin Aggregated Logging
+
+## Building on OpenShift
+Choose the project you want to hold your logging infrastructure. It can be
+any project.
+
+Instantiate the [dev-builds template](hack/templates/dev-builds.yaml)
+to define BuildConfigs for all images and ImageStreams to hold their
+output. You can do this before or after deployment, but before is
+recommended. A logging deployment defines the same ImageStreams, so it
+is normal to see errors about already-defined ImageStreams when building
+from source and deploying. Normally existing ImageStreams are deleted
+at installation to enable redeployment with different images. To prevent
+your customized ImageStreams from being deleted, ensure that they are not
+labeled with `logging-infra=support` like those generated by the deployer.
+
+The template has parameters to specify the repository and branch to use
+for the builds. The defaults are for origin master. To develop your own
+images, you can specify your own repos and branches as needed.
+
+A word about the openshift-auth-proxy: it depends on the "node" base
+image, which is intended to be the DockerHub nodejs base image. If you
+have defined all the standard templates, they include a nodejs builder image
+that is also called "node", and this will be used instead of the intended
+base image, causing the build to fail. You can delete it to resolve this
+problem:
+
+    oc delete is/node -n openshift
+
+The builds should start once defined; if any fail, you can retry them with:
+
+    oc start-build <component>
+
+e.g.
+
+    oc start-build openshift-auth-proxy
+
```

Comment:
"This will start the builds in the background.  Use `oc logs -f build/<component>` to follow the build logs, or use `oc start-build --follow <component>` to run the build in the foreground and follow the build logs."